compiler/parser/statement_parsing.go

```diff
@@ -258,7 +258,7 @@ func (p *Parser) parseBlockStatement() *ast.BlockStatement {
 		p.nextToken()
 	}
 
-	for !p.curTokenIs(token.End) && !p.curTokenIs(token.Else) && !p.curTokenIs(token.ElsIf) {
+	for !p.curTokenIs(token.End) && !p.curTokenIs(token.Else) && !p.curTokenIs(token.ElsIf) && !p.curTokenIs(token.When) {
```

Comment:
I know this part is not good since block parsing stops whenever parser encounters those tokens. I'm thinking about giving tokens as arguments, which tell where to stop parsing block statement. I'll work on it in next PR.

Reply:
I think you can first open a PR to refactor this part's logic if you could.